iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -591,6 +648,12 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
     [self sendOutcomes:action.outcomes forMessageId:message.messageId];
 }
 
+- (void)messageViewDidDisplayPage:(OSInAppMessage *)message withPageId:(NSString *)pageId {
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
just curious why this method needs dispatch and messageViewImpressionRequest not?

Reply:
It doesn't need it but for these requests we don't need to wait for the result to start popping back up the stack